### ISSUE TITLE
I would like to have an animation to desaturate elements when the window is not focused.

### ISSUE THREAD

**Issue by 73** [BrandonKirbyson/VSCode-Animations#16]
Something along the lines of:
```js

let dimmable = ".sidebar, .activitybar, .breadcrumbs, .statusbar";

window.addEventListener("blur", function () {
  for (element of document.querySelectorAll(dimmable)) {
    element.style.transition = "opacity 0.2s";
    element.style.opacity = 0.6;
  }
});

window.addEventListener("focus", function () {
  for (element of document.querySelectorAll(dimmable)) {
    element.style.transition = "opacity 0.2s";
    element.style.opacity = 1.0;
  }
});

```

Which should desaturate everything except the editor. I find it useful to be able to read the source displayed in an unfocussed window. 

Thanks,
73

**Comment by BrandonKirbyson** [comment]
Ok, this is a good idea actually. Are you thinking absolutely everything except editor or editor and terminal or something. Also I will probably add this in with the default setting of fading everything on blur.

**Comment by BrandonKirbyson** [comment]
Ok, this should be implemented now! Here are the settings:
![image](https://github.com/BrandonKirbyson/VSCode-Animations/assets/86384607/586be809-a30e-45ba-94af-73259ba7f6b7)
And you can change the duration of course:
![image](https://github.com/BrandonKirbyson/VSCode-Animations/assets/86384607/feea1a46-690a-4d4a-948c-585c6df633ad)
This is now published in version `1.0.13`

**Comment by 73** [comment]
Just tested it. Looks amazing. Thank you.

**Comment by BrandonKirbyson** [comment]
Great, thanks for idea

**Comment by 73** [comment]
It would be nice if the user could provide additional CSS selectors. I would like to apply the effect to my breadcrumbs and status bar as well.

**Comment by BrandonKirbyson** [comment]
Ok, I can add that soon. Would it be better if I just made sure that it selected everything but the editor which I could do with a css selector ex: `.split-view:not(:has(.editor))` since people will probably want everything selected? Maybe I can still add an option to change what selectors stay in focus (undimmed) so like the inverse of what you are suggesting?

Let me know what you think is the best way to add this but any way is easily possible.

**Comment by 73** [comment]
Thank you for getting back to me that quickly.

I am fine either way. I am not sure what the majority of people want. I am sure some want event to dim the editor pane. As I am sometimes working in two windows side by side, bringing code form one project to another, I would like to see the editor pane clearly but have the active window distinguished from the inactive a very obvious way. Hence, I would love to dim *everything but the editor pane*. 

There is also [this issue](https://github.com/microsoft/vscode/issues/30522), where people want to dim everything but the focussed pane. It is really up to preference, I guess.

**Comment by BrandonKirbyson** [comment]
I just made a commit that adds all this but probably won't publish it until later tomorrow since there is another feature that should be added before the next version. Thanks for the great ideas and for starring the repo!

**Comment by BrandonKirbyson** [comment]
Published `1.0.14`!

**Comment by 73** [comment]
That looks *very* nice. Thank you!
